### PR TITLE
runner args

### PR DESCRIPTION
**Pull request by ditwoo**
## Before submitting (checklist)

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)?
- [ ] Did you check the code style? `catalyst-make-codestyle && catalyst-check-codestyle` (`pip install -U catalyst-codestyle`).
- [ ] Did you make sure to update the docs? We use Google format for all the methods and classes.
- [ ] Did you check the docs with `make check-docs`?
- [ ] Did you write any new necessary tests?
- [ ] Did you check that your code passes the unit tests `pytest .` ? 
- [ ] Did you add your new functionality to the docs?
- [ ] Did you update the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

PS
- [x] I know, that I could [join slack](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for pull request discussion.

### CODE REVIEW

File: catalyst/runners/runner.py [hunk 1]
```diff
@@ -32,9 +37,26 @@
...
+            return DistributedDataParallelAMPEngine()
+        elif fp16 and has_multiple_gpus:
+            return DataParallelAMPEngine()
+        elif fp16 and NUM_CUDA_DEVICES == 1:
+            return AMPEngine()
+        elif ddp and has_multiple_gpus:
+            return DistributedDataParallelEngine()
+        elif has_multiple_gpus:
+            return DataParallelEngine()
+    return DeviceEngine("cuda")
```
Comment (Scitator) [inline catalyst/runners/runner.py]: what about apex?)

**Review by Scitator — DISMISSED** [review]

File: catalyst/runners/runner.py [hunk 2]
```diff
@@ -32,9 +37,26 @@
 from catalyst.utils.data import get_loaders_from_params
 from catalyst.utils.misc import maybe_recursive_call, set_global_seed
 
+if SETTINGS.amp_required:
+    from catalyst.engines import AMPEngine, DataParallelAMPEngine, DistributedDataParallelAMPEngine
+
 
 def _get_default_engine(fp16: bool = False, ddp: bool = False):
```
Comment (Scitator) [inline catalyst/runners/runner.py]: btw, could we move it to `catalyst/utils/torch.py` as `get_engine`?
like https://github.com/catalyst-team/catalyst/blob/dev/catalyst/utils/torch.py#L170

Comment (ditwoo) [reply catalyst/runners/runner.py]: sure

File: catalyst/runners/runner.py (hunk 1 above)
Comment (ditwoo) [reply catalyst/runners/runner.py]: added

**Pull request merged**